packages/cosmic-swingset/lib/ag-solo/vats/bootstrap.js

```diff
@@ -66,9 +68,58 @@ export function buildRootObject(vatPowers, vatParameters) {
       E(vats.priceAuthority).makePriceAuthority(),
     ]);
 
+    const {
+      nameHub: agoricNames,
+      nameAdmin: agoricNamesAdmin,
+    } = makeNameHubKit();
+    const {
+      nameHub: namesByAddress,
+      nameAdmin: namesByAddressAdmin,
+    } = makeNameHubKit();
+
+    async function installEconomy() {
+      // Create a mapping from all the nameHubs we create to their corresponding
+      // nameAdmin.
+      const nameAdmins = makeStore();
+      await Promise.all(
+        ['brand', 'installation', 'issuer', 'instance', 'uiConfig'].map(
+          async nm => {
+            const { nameHub, nameAdmin } = makeNameHubKit();
+            await E(agoricNamesAdmin).update(nm, nameHub);
+            nameAdmins.init(nameHub, nameAdmin);
+          },
+        ),
+      );
+
+      // Install the economy, giving it access to the name admins we made.
+      await installEconomyOnChain({
+        agoricNames,
+        board,
+        chainTimerService,
+        nameAdmins,
+        priceAuthority,
+        zoe,
+      });
+    }
+
+    // Now we can bootstrap the economy!
+    await installEconomy();
+    const centralIssuer = await E(agoricNames).lookup('issuer', 'Moe');
```

Comment:
Would it be better to get the issuer from the treasury itself?

Reply:
Also could this be in a Promise.all with the line before it?

Reply:
> Would it be better to get the issuer from the treasury itself?

Maybe.  I'm trying to ensure that the same thing we put into the agoricNames is used during bootstrap, which indirectly verifies that the entry is correct.

> Also could this be in a Promise.all with the line before it?

It can't be in a `Promise.all` unless we reserve `Moe`.  Maybe we should.